R/pipeline.R

```diff
@@ -0,0 +1,221 @@
+    split = TRUE
+  )
+  sink(
+    logfile_connection,
+    type = "message",
+    append = TRUE
+  )
+  on.exit(sink(file = NULL))
+  cli::cli_alert_info("Starting run at {Sys.time()}")
+  cli::cli_alert_info("Using job id {.field {config[['job_id']]}}")
+  cli::cli_alert_info("Using task id {.field {config[['task_id']]}}")
+
+  # Errors within `process_pipeline()` are downgraded to warnings so
+  # they can be logged and stored in Blob. If there is an error,
+  # `pipeline_success` is set to false, which will be stored in the
+  # metadata in the next PR.
+  pipeline_success <- rlang::try_fetch(
+    process_pipeline(config, output_dir),
+    error = function(con) {
+      cli::cli_warn("Pipeline run failed",
+        parent = con,
+        class = "Run_failed"
+      )
+      FALSE
+    }
+  )
+
+  # TODO: Move metadata to outer wrapper
+  cli::cli_alert_info("Finishing run at {Sys.time()}")
+  invisible(pipeline_success)
+}
+
+#' Run the Model Fitting Process
+#'
+#' @param config A list containing configuration settings for the pipeline,
+#' including paths to data, exclusions, disease parameters, model settings,
+#' and other necessary inputs.
+#'
+#' @details
+#' This function performs the core model fitting process within the Rt
+#' estimation pipeline, including reading data, applying exclusions, fitting
+#' the model, and writing outputs such as model samples, summaries, and logs.
+#'
+#' @return Returns `TRUE` on success. Errors are caught by the outer pipeline
+#' logic and logged accordingly.
+#'
+#' @rdname pipeline
+#' @export
+process_pipeline <- function(config, output_dir) {
+  cases_df <- read_data(
+    data_path = config[["data"]][["path"]],
+    disease = config[["disease"]],
+    state_abb = config[["geo_value"]],
+    report_date = config[["report_date"]],
+    max_reference_date = config[["max_reference_date"]],
+    min_reference_date = config[["min_reference_date"]]
+  )
+
+  if (!rlang::is_null(config[["exclusions"]][["path"]])) {
+    exclusions_df <- read_exclusions(config[["exclusions"]][["path"]])
+    cases_df <- apply_exclusions(cases_df, exclusions_df)
+  } else {
+    cli::cli_alert("No exclusions file provided. Skipping exclusions")
+  }
+
+  params <- read_disease_parameters(
+    generation_interval_path = config[["parameters"]]
+    [["generation_interval"]]
+    [["path"]],
+    delay_interval_path = config[["parameters"]]
+    [["delay_interval"]]
+    [["path"]],
+    right_truncation_path = config[["parameters"]]
+    [["right_truncation"]]
+    [["path"]],
+    disease = config[["disease"]],
+    as_of_date = config[["as_of_date"]],
+    group = config[["geo_value"]],
+    report_date = config[["report_date"]]
+  )
+
+  fit <- fit_model(
+    data = cases_df,
+    parameters = params,
+    seed = config[["seed"]],
+    horizon = config[["horizon"]],
+    priors = config[["priors"]],
+    sampler_opts = config[["sampler_opts"]]
+  )
+  diagnostics <- extract_diagnostics(
```

Comment:
```suggestion

  diagnostics <- extract_diagnostics(
```

And others?

Reply:
Wait I'm not following the suggested diff here. What's the change?

Reply:
Newline spacing

Reply:
https://github.com/CDCgov/cfa-epinow2-pipeline/pull/79/commits/04b206e814eb649e596d8c91b274735e73283885